app/scripts/model/dataProxy.js

```diff
@@ -340,28 +357,39 @@
             _osSurvivalAttrMeta.display_name = 'Overall Survival';
             _osSurvivalAttrMeta.filter = [];
             _osSurvivalAttrMeta.show = true;
-            _patientAttributes.unshift(_osSurvivalAttrMeta);
+            _osSurvivalAttrMeta.priority = 1;
+            _patientAttributes[_osSurvivalAttrMeta.attr_id] = _osSurvivalAttrMeta;
           }
 
           // add Cancer Study
           if (self.getCancerStudyIds().length > 1) {
-            _patientAttributes.unshift({
+            _patientAttributes['study_id'] = {
```

Comment:
I thought about it. Not really necessary. It's nice to have but it needs couple few loops. These attributes are mostly designed specifically, and normally they are in preSelectedAttribtues or in clinicalAttrsPriority list. And I have https://github.com/cBioPortal/iViz/pull/178/commits/fe301e449a71bec63153f08c6fe05f5e8be884a9#diff-5baab798025ed50b358d0b2e1421cbe3R687 to deal with attributes without keys or numOfDatum